service/src/test/java/uk/gov/hmcts/reform/fpl/service/NoticeOfProceedingsTemplateDataGenerationServiceTest.java

```diff
@@ -0,0 +1,238 @@
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(ImmutableList.of(
+                Element.<Child>builder()
+                    .id(UUID.randomUUID())
+                    .value(Child.builder()
+                        .party(ChildParty.builder()
+                            .firstName("Bran")
+                            .lastName("Stark")
+                            .build())
+                        .build())
+                    .build()))
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
+            .noticeOfProceedings(NoticeOfProceedings.builder()
+                .judgeAndLegalAdvisor(createJudgeAndLegalAdvisor())
+                .proceedingTypes(emptyList())
+                .build())
+            .build();
+
+        DocmosisNoticeOfProceeding templateData = noticeOfProceedingsTemplateDataGenerationService
+            .getTemplateData(caseData);
+        assertThat(templateData.getChildrenNames()).isEqualTo("Bran Stark");
+    }
+
+    @Test
+    void shouldFormatMagistrateFullNameWhenJudgeTitleIsSetToMagistrate() {
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(children)
+            .noticeOfProceedings(NoticeOfProceedings.builder()
+                .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder()
+                    .judgeTitle(MAGISTRATES)
+                    .judgeFullName("James Nelson")
+                    .build())
+                .proceedingTypes(emptyList())
+                .build())
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
+            .build();
+
+        DocmosisNoticeOfProceeding templateData = noticeOfProceedingsTemplateDataGenerationService
+            .getTemplateData(caseData);
+        assertThat(templateData.getJudgeTitleAndName()).isEqualTo("James Nelson (JP)");
+    }
+
+    @Test
+    void shouldSetJudgeTitleAndNameToEmptyStringWhenJudgeTitleAndNameIsEmpty() {
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(children)
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
+            .noticeOfProceedings(NoticeOfProceedings.builder()
+                .judgeAndLegalAdvisor(JudgeAndLegalAdvisor.builder().build())
+                .proceedingTypes(emptyList())
+                .build())
+            .build();
+        DocmosisNoticeOfProceeding templateData = noticeOfProceedingsTemplateDataGenerationService
+            .getTemplateData(caseData);
+        assertThat(templateData.getJudgeTitleAndName()).isEqualTo("");
+    }
+
+    @Test
+    void shouldReturnFirstApplicantNameWhenMultipleApplicantsArePresent() {
+        CaseData caseData = initNoticeOfProceedingCaseData()
+            .children1(children)
+            .orders(Orders.builder()
+                .orderType(ImmutableList.of(CARE_ORDER)).build())
```

Comment:
minor: same here, we can use `Collections.singletonList` instead of `ImmutableList.of` as its a single value in the list. Looking at it again, looks like we can have this in  `initNoticeOfProceedingCaseData()`?

Reply:
agree, done.